crates/stateless/src/validation.rs

```diff
@@ -153,20 +151,20 @@ where
 /// implementation of the `StatelessTrie` for custom trie backends or optimizations.
 ///
 /// See `stateless_validation` for detailed documentation of the validation process.
-pub fn stateless_validation_with_trie<T, ChainSpec, E>(
-    current_block: Block,
+pub fn stateless_validation_with_trie<T, B, ChainSpec, E>(
+    current_block: B,
     witness: ExecutionWitness,
     chain_spec: Arc<ChainSpec>,
     evm_config: E,
 ) -> Result<B256, StatelessValidationError>
 where
     T: StatelessTrie,
+    B: RecoverableBlock<Block>,
     ChainSpec: Send + Sync + EthChainSpec<Header = Header> + EthereumHardforks + Debug,
     E: ConfigureEvm<Primitives = EthPrimitives> + Clone + 'static,
 {
-    let current_block = current_block
-        .try_into_recovered()
-        .map_err(|err| StatelessValidationError::SignerRecovery(Box::new(err)))?;
+    let current_block =
+        current_block.try_into_recovered().map_err(|_| StatelessValidationError::SignerRecovery)?;
```

Comment:
I don't think this makes a lot of sense because this is more or less same as accepting

```
current_block: RecoveredBlock<B>,
```

just with extra steps

can we accept recoveredblock here instead?